components/MentorRegistration/components/PersonDetails.js

```diff
@@ -59,8 +61,64 @@ function PersonDetails({
     },
   ];
 
+  const [showImageCropper, setShowImageCropper] = useState(false);
+  const [imageSrc, setImageSrc] = useState("");
+  const [fileName, setFileName] = useState("");
+
+  const convertBase64 = (file) => {
+    return new Promise((resolve, reject) => {
+      const fileReader = new FileReader();
+      fileReader.readAsDataURL(file);
+
+      fileReader.onload = () => {
+        resolve(fileReader.result);
+      };
+
+      fileReader.onerror = (error) => {
+        reject(error);
+      };
+    });
+  };
+
+  const handleImageChange = async (e) => {
+    const file = e.target.files[0];
+    if (!file) return;
+    setShowImageCropper(true);
+    const base64 = await convertBase64(file);
+    setFileName(file.name);
+    setImageSrc(base64);
+  };
+
+  const handleImageCropperChange = async (imageSrc) => {
+    setShowImageCropper(false);
+    if (!imageSrc) return;
+    setImageSrc(imageSrc);
+    const imageClouindaryUrl = await uploadToCloudinary(imageSrc);
+    handleUploadImageChange(fileName, imageClouindaryUrl);
+  };
+
+  const uploadToCloudinary = async (imageSrc) => {
+    const blob = await fetch(imageSrc).then((r) => r.blob());
```

Comment:
Don't mix `await` and `then` inside a function. It leads to buggy and hard-to-read code. Please use `await` here.

Reply:
Hey @glokta1 i have made the changes